### ISSUE TITLE
Allow plugins and parsers to initialize configuration

### ISSUE THREAD

**Issue by manuth** [eslint/eslint#12630]
Hey guys
So I've been working with `@typescript-eslint` for quite a while and I've noticed that if I'm using the `parserOptions.project` setting which points at a `tsconfig.json` `eslint` acts differently depending in which directory I am.

The reason is because the path to the `tsconfig.json`-file (like, for example `./src/App/tsconfig.json` isn't resolved to its absolute path on initialization.

When using `.eslintrc.js` you could simply replace the setting by `path.join(__dirname, "./src/App/tsconfig.json");` but, that's not possible in JSON or YAML.

I would recommend that plugins and parsers should be able to initialize the settings when the configuration is loaded by passing the settings to them.

https://github.com/eslint/eslint/blob/ca3b2a62c9e829dc4534bca3643d6bc729b46df0/lib/cli-engine/config-array-factory.js#L661-L670

***Example:***
```js
for (let plugin of plugins) {
    if (plugin.loadConfig) {
        plugin.loadConfig(filePath, config);
  }
}
```

**Comment by kaicataldo** [comment]
Thanks for the proposal. We ask that all changes to core go through the RFC process outlined [here](https://github.com/eslint/rfcs).

There's an [existing RFC](https://github.com/eslint/rfcs/pull/42#issuecomment-561296926) with discussion that I think would solve this indirectly. It doesn't exist as a separate RFC yet, but if you'd like to create one, we'd love to discuss it with you!
